### ISSUE TITLE
GPs office: doctors and handymen stuck / crash

### ISSUE THREAD

**Issue by maverique** [CorsixTH/CorsixTH#1051]
So basically some times a GP's office will "hang", and a doctor will be inside typing at the computer, then getting up, then sitting down again, over and over and over. If I pick him up and drop him elsewhere another doctor will try to come in but fail. After a while, the game crashes.

I'm not sure if this is the same issue, but usually at the same time a handyman will get stuck a couple of tiles away from the door to that GP. He won't move, he won't do anything. Pick him up and he'll walk back to the same spot and get stuck straight away.

The game doesn't crash to desktop, it throws a message asking if I want to attempt a recovery and if I say no, everything freezes. If I say yes, the game will go for 5 or so more seconds and throw another error.

Currently on this save I'm uploading I'm like 10 seconds away from the crash. This is the info the log spits out and if there's anything else I can do to help recreate this bug, just ask. I'm playing on Windows 10 x64, on the GOG version, and these bugs happen to me on 0.50, 0.60b1 and 0.60b2. I know basically nothing of what the error log is saying as I don't code ;)

[Save file.](https://drive.google.com/file/d/0B9_mh8vFFnvVQ1d0dW0tekQwUHc/view?usp=sharing)

Checking for CorsixTH updates...
You are running the latest version of CorsixTH.
An error has occurred!
Almost anything can be the cause, but the detailed information below can help the developers find the source of the error.
Running: The timer handler.
A stack trace is included below, and the handler has been disconnected.
...Games\Theme Hospital\CorsixTH\Lua\rooms\slack_tongue.lua:56: attempt to index a nil value (local 'slicer')
stack traceback:
...Games\Theme Hospital\CorsixTH\Lua\rooms\slack_tongue.lua:56: in method 'commandEnteringPatient'
...\GalaxyClient\Games\Theme Hospital\CorsixTH\Lua\room.lua:400: in method 'onHumanoidEnter'
...es\Theme Hospital\CorsixTH\Lua\humanoid_actions\walk.lua:356: in field 'on_next_tile_set'
...es\Theme Hospital\CorsixTH\Lua\humanoid_actions\walk.lua:150: in function <...es\Theme Hospital\CorsixTH\Lua\humanoid_actions\walk.lua:137>
(...tail calls...)
...alaxyClient\Games\Theme Hospital\CorsixTH\Lua\entity.lua:219: in method 'tick'
...GalaxyClient\Games\Theme Hospital\CorsixTH\Lua\world.lua:1064: in method 'onTick'
...)\GalaxyClient\Games\Theme Hospital\CorsixTH\Lua\app.lua:979: in function <...)\GalaxyClient\Games\Theme Hospital\CorsixTH\Lua\app.lua:976>
(...tail calls...)
...)\GalaxyClient\Games\Theme Hospital\CorsixTH\Lua\app.lua:880: in function <...)\GalaxyClient\Games\Theme Hospital\CorsixTH\Lua\app.lua:875>

**Comment by Alberth289346** [comment]
The log is actually a stack trace, a list of "filename : line-number : what it was doing there" lines.

If you compare the lines with the log in #694, they look the same to me, so as for the log it looks like a duplicate issue, but you seem to be talking about something else?

**Comment by maverique** [comment]
Yeah I posted on that one but I'm not entirely sure that me and the original poster are referencing the exact same thing so I made this issue here.

**Comment by TheCycoONE** [comment]
I didn't reproduce the crash with the slack tongue on the latest master, but as mentioned that's not the issue here. The GP room is already broken at the start of the save.

**Comment by maverique** [comment]
If there's anything I can do to help you track down and kill this nasty bug, just ask.

**Comment by TheCycoONE** [comment]
A couple questions: did you notice this bug before you saved, or only after loading the save? Was there anything else in the original game log? And do you have an earlier save from when the GP still worked that eventually reproduces this?

**Comment by TheCycoONE** [comment]
Duplicate of #76
